drivers/src/hmac384.rs

```diff
@@ -212,17 +212,30 @@ impl Hmac384 {
                 .map_err(|err| err.into_write_tag_err()),
         };
 
-        self.zeroize();
+        self.zeroize_internal();
 
         result
     }
 
     /// Zeroize the hardware registers.
-    pub fn zeroize(&mut self) {
+    fn zeroize_internal(&mut self) {
         self.hmac.regs_mut().ctrl().write(|w| w.zeroize(true));
     }
+
+    /// Zeroize the hardware registers.
+    ///
+    /// This is useful to call from a fatal-error-handling routine.
+    ///
+    /// # Safety
+    ///
+    /// This function is safe to call from a trap handler.
```

Comment:
Can you add "Caller must be certain that the results of any pending cryptographic operations will not be used after this function is called" here and elsewhere, to make it clear that this should only be used in "fatal-error" type scenarios.

Reply:
actually, I'll do this in a PR